02-deploying-an-app/005-add-aws-resources.md

```diff
@@ -0,0 +1,56 @@
+# Adding AWS resources to your environment
+
+Through the [cloud-platform-environments](https://github.com/ministryofjustice/cloud-platform-environments/) repository, you can provision AWS resources for your environments. This is done using terraform and more specifically, terraform modules we provide for use on the Cloud Platform.
+
+The documentation for the modules lives in each module's repository and you can find a list of the available ones below.
+
+## Available modules
+
+- [ECR Repository](https://github.com/ministryofjustice/cloud-platform-terraform-ecr-credentials)
+- [S3 Bucket](https://github.com/ministryofjustice/cloud-platform-terraform-s3-bucket)
+- [RDS instance](https://github.com/ministryofjustice/cloud-platform-terraform-rds-instance)
+- [Elasticache Redis Cluster](https://github.com/ministryofjustice/cloud-platform-terraform-elasticache-cluster)
+- [DynamoDB Table](https://github.com/ministryofjustice/cloud-platform-terraform-dynamodb-cluster)
+- [SNS Topic](https://github.com/ministryofjustice/cloud-platform-terraform-sns-topic)
+
+## Usage
+
+In each terraform module repository, you will find a directory named `example` which includes sample configuration for use in Cloud Platform.
+
+In your namespace's path in the [cloud-platform-environments](https://github.com/ministryofjustice/cloud-platform-environments/) repository, create a directory called `resources` (if you have not created one already) and refer to the module's example to define your resources.
+
+Each example will have some global configuration defined, however, this should only be declared once, regardless of the number of modules used:
+```
+terraform {
+  backend "s3" {}
+}
+
+provider "aws" {
+  region = "eu-west-1"
+}
+```
+
+Additionally, some example might define variables; again, these should only be declared once per namespace:
+```
+variable "cluster_name" {}
+
+variable "cluster_state_bucket" {}
+```
+
+The main README file of each module repository will list all the available configuration options that can be passed to the module.
+
```

Comment:
I think we need to have a section somewhere here that explains how we do the secret -- i.e. how it gets to be a kubernetes secret. I know that you've made this consistent across all of the examples in the modules (big 👍 for that) but the idea might not be entirely clear to people so a few lines here would be helpful I think.

What do you think?

Reply:
How do you feel about bda629e ?

Reply:
Yep perfect.